meson.build

```diff
@@ -0,0 +1,87 @@
+project('macutils', 'c')
+# Use the following flags on the CF macro definition as needed.
+#
+# -DBSD if you are on a BSD system
+#
+# -DTYPES_H if your system has /usr/include/sys/types.h
+#
+# -DDIRENT_H if your system has /usr/include/dirent.h
+#
+# -DTERMIOS_H if your system has /usr/include/sys/termios.h
+#
+# -DNODOT if you do not want to create files with an initial period
+#
+# -DLATIN1 if your system supports LATIN-1 and you want to use it
+#
+# Note you can use at most one of the following four!
+#
+# -DNOMKDIR if your system does not have the mkdir system call
+#
+# -DAUFS if you want to use an AUFS file system
+#
+# -DAUFSPLUS if you use CAP 6.0 and want to use times on files
+#
+# -DAPPLEDOUBLE if you want to be able to use an AppleDouble file system
+#
+add_global_arguments([
+    '-DNODOT',
+    '-DAPPLEDOUBLE'],
+    language: 'c')
+add_global_arguments(['-Wall'], language: 'c')
```

Comment:
-Wall is set by default, in accordance with Meson's builtin `-Dwarning_level=1` option. Level 2 is -Wextra, level 0 is nothing.

It's usually best to let Meson handle this directly, because it can adapt to the compiler automatically, although the main advantage there is for cross-platform code that needs to handle Windows compilers as well... Meson still warns about it as a matter of style though since it doesn't know whether it's safe to assume gcc/clang.

Reply:
Oh thanks, didn't know. I'm very new to using meson (I used this project as a learning opportunity).

Reply:
No problem :) always happy to hand out tips and help people use Meson more effectively.